io/src/openni2_grabber.cpp

```diff
@@ -633,22 +633,22 @@ pcl::io::OpenNI2Grabber::convertToXYZRGBPointCloud (const Image::Ptr &image, con
   float fx_inv = 1.0f / fx;
   float fy_inv = 1.0f / fy;
 
-  const std::uint16_t* depth_map = (const std::uint16_t*) depth_image->getData ();
+  auto* depth_map = reinterpret_cast<const std::uint16_t*>(depth_image->getData ());
```

Comment:
The `const_cast` is strange. How about we use `const auto* depth_map = ...`, apply `fillDepthImageRaw` to `depth_resize_buffer_.data()` and then assign `depth_map = depth_resize_buffer_.data();`? (3 more instances below for rgb_buffer, depth_map, and ir_map)